### DISCUSSION TITLE
Access via domain name

### DISCUSSION THREAD

**Discussion by Servis-Chomutov** [Part-DB/Part-DB-server#365] [Q&A] [answered]
Hello, 

I have successfully installed a Part-DB via docker on the Ubuntu Server. I am now struggling with setting IP addresses. Could someone help me? 

Part-DB is now running just fine and is accessible via the local IP address (192.168.1.20:8080). However, before I started making labels for all the parts and storage units, I realized that the QR code would need to be regenerated if the IP changes in the future (and all the labels printed again.)

To avoid this, I would like to set the Part-DB to be accessible via our domain. In case we change the ISP or our local IP will change, the QR codes will still work. But I am not sure how to correctly do this. 

Here are the details about the current setup. 

a) PartDB is installed on a virtual machine, that is physically hosted at my location. The machine is running Proxmox and Part-DB is installed on Ubuntu Server. The local IP address is 192.168.1.20. 

b) I have a public IP address available. 

c) I would like to use our domain (www.servis-chomutov.cz) to access the Part-DB from the outside. 

I am assuming that I will need to make a subdomain (like partdb.servis-chomutov.cz), redirect the DNS to our public IP, and set NAT on the router to forward it to the local IP. But I am not sure about it and would like to ask someone more experienced. 

Thanks anyone for helping me out!

**Comment by jbtronics** [comment, marked as answer]
Yes basically that should work that way. If you want to expose Part-DB to the internet (either on the web server Part-DB or by running a reverse proxy in front of it).
And if you don't need it public accessible, then you can just configure a local domain in your router (or even the hosts file of your PC), with a domain like partdb.lan


In principle you can also still use QR-codes with an old domain, when you use the scanner built-in in Part -DB (it will redirect you to the correct part no matter what the domain in the QR code is). If you want to scan it directly with some normal QR code scanner app it will not work, if the url is not available anymore.

**Reply by Servis-Chomutov** [reply]
Hello, thank you, I set it up and it works correctly. I was confused with the QR code generator, now I know how it works. 
So for anyone who wants to do the same thing, here is what I did:

1. Set a DNS record for your subdomain with your domain provider. So for example sub.domain.com goes to your public IP
2. On your router set port forwarding from port 80 to the port and IP that your server with part-db runs on (for example 192.168.1.10:8123)
3.  You are all set! 

I am now playing with some proxy server to be able to use my public IP for hosting multiple services, but I haven't figured that out yet, I will do it when i need it.

**Comment by w0hc** [comment]
I can't seem to access my install from a domain name. I'm using a cloudflare tunnel, and the page appears, but all the css is missing, and there's multiple errors in the browser console.

`login:24 Failed to find a valid digest in the 'integrity' attribute for resource 'https://part-db.my-domain.net/build/1381.097e9337.css' with computed SHA-384 integrity 'PrzjRAlcf6i0TxvG1uII5/eMxuStZiMGN1bKlBIbmzS0e5UMRdihqYIvDAYkHb8Q'. The resource has been blocked.
login:25 Failed to find a valid digest in the 'integrity' attribute for resource 'https://part-db.my-domain.net/build/8919.ee2af1da.css' with computed SHA-384 integrity 'y/0A3cgdEHcdHEQOLWUSCxWpiX5JI9IAH7oVpvlG7KPc7tE1tSWGOsNBt1iZn0Mu'. The resource has been blocked.
login:25 Failed to find a valid digest in the 'integrity' attribute for resource 'https://part-db.my-domain.net/build/app.a3b7998e.css' with computed SHA-384 integrity 'ms82yf/pMguY6JGFa2GSwG9BDfFVnnE5+mWI8Z1Qtg/U9DkTBXirD3pPwn0VOC+U'. The resource has been blocked.
login:1 Failed to find a valid digest in the 'integrity' attribute for resource 'https://part-db.my-domain.net/build/8919.a05c51b2.js' with computed SHA-384 integrity 'v42CA6NIrTNcuQTJE0QfACkp9XPrkDV0izqnW7dFKeCGSWzV7SiH1vP7BAGiqiIA'. The resource has been blocked.
login:1 Failed to find a valid digest in the 'integrity' attribute for resource 'https://part-db.my-domain.net/build/6493.514c1e81.js' with computed SHA-384 integrity 'hSwpAetGShRqzTX7VuvvT8ILN0tMF4SoY9I//Gg+L1bVR6Cfr9qrR9ouULv4ph1f'. The resource has been blocked.`

**Reply by jbtronics** [reply]
This is most likely caused by cloudflare modifying ("optimizing") the assets served by Part-DB. You should check your cloudflare settings: https://www.lkhrs.com/blog/2021/06/hugo-css-error/ and https://stackoverflow.com/a/70872265

**Reply by w0hc** [reply]
Thanks! I've changed the Cloudflare settings and I'll see if that fixes it.
